components/eamxx/src/physics/rrtmgp/eamxx_rrtmgp_process_interface.cpp

```diff
@@ -955,67 +976,67 @@ void RRTMGPRadiation::run_impl (const double dt) {
-      auto sw_flux_up_k      = subview_2dk(m_buffer.sw_flux_up_k);
-      auto sw_flux_dn_k      = subview_2dk(m_buffer.sw_flux_dn_k);
-      auto sw_flux_dn_dir_k  = subview_2dk(m_buffer.sw_flux_dn_dir_k);
-      auto lw_flux_up_k      = subview_2dk(m_buffer.lw_flux_up_k);
-      auto lw_flux_dn_k      = subview_2dk(m_buffer.lw_flux_dn_k);
-      auto sw_clnclrsky_flux_up_k      = subview_2dk(m_buffer.sw_clnclrsky_flux_up_k);
-      auto sw_clnclrsky_flux_dn_k      = subview_2dk(m_buffer.sw_clnclrsky_flux_dn_k);
-      auto sw_clnclrsky_flux_dn_dir_k  = subview_2dk(m_buffer.sw_clnclrsky_flux_dn_dir_k);
-      auto sw_clrsky_flux_up_k      = subview_2dk(m_buffer.sw_clrsky_flux_up_k);
-      auto sw_clrsky_flux_dn_k      = subview_2dk(m_buffer.sw_clrsky_flux_dn_k);
-      auto sw_clrsky_flux_dn_dir_k  = subview_2dk(m_buffer.sw_clrsky_flux_dn_dir_k);
-      auto sw_clnsky_flux_up_k      = subview_2dk(m_buffer.sw_clnsky_flux_up_k);
-      auto sw_clnsky_flux_dn_k      = subview_2dk(m_buffer.sw_clnsky_flux_dn_k);
-      auto sw_clnsky_flux_dn_dir_k  = subview_2dk(m_buffer.sw_clnsky_flux_dn_dir_k);
-      auto lw_clnclrsky_flux_up_k      = subview_2dk(m_buffer.lw_clnclrsky_flux_up_k);
-      auto lw_clnclrsky_flux_dn_k      = subview_2dk(m_buffer.lw_clnclrsky_flux_dn_k);
-      auto lw_clrsky_flux_up_k      = subview_2dk(m_buffer.lw_clrsky_flux_up_k);
-      auto lw_clrsky_flux_dn_k      = subview_2dk(m_buffer.lw_clrsky_flux_dn_k);
-      auto lw_clnsky_flux_up_k      = subview_2dk(m_buffer.lw_clnsky_flux_up_k);
-      auto lw_clnsky_flux_dn_k      = subview_2dk(m_buffer.lw_clnsky_flux_dn_k);
-      auto sw_bnd_flux_up_k  = subview_3dk(m_buffer.sw_bnd_flux_up_k);
-      auto sw_bnd_flux_dn_k  = subview_3dk(m_buffer.sw_bnd_flux_dn_k);
-      auto sw_bnd_flux_dir_k = subview_3dk(m_buffer.sw_bnd_flux_dir_k);
-      auto sw_bnd_flux_dif_k = subview_3dk(m_buffer.sw_bnd_flux_dif_k);
-      auto lw_bnd_flux_up_k  = subview_3dk(m_buffer.lw_bnd_flux_up_k);
-      auto lw_bnd_flux_dn_k  = subview_3dk(m_buffer.lw_bnd_flux_dn_k);
-      auto sfc_flux_dir_vis_k = subview_1dk(m_buffer.sfc_flux_dir_vis_k);
-      auto sfc_flux_dir_nir_k = subview_1dk(m_buffer.sfc_flux_dir_nir_k);
-      auto sfc_flux_dif_vis_k = subview_1dk(m_buffer.sfc_flux_dif_vis_k);
-      auto sfc_flux_dif_nir_k = subview_1dk(m_buffer.sfc_flux_dif_nir_k);
-      auto aero_tau_sw_k     = subview_3dk(m_buffer.aero_tau_sw_k);
-      auto aero_ssa_sw_k     = subview_3dk(m_buffer.aero_ssa_sw_k);
-      auto aero_g_sw_k       = subview_3dk(m_buffer.aero_g_sw_k);
-      auto aero_tau_lw_k     = subview_3dk(m_buffer.aero_tau_lw_k);
+      // Note, ncol will not necessary be m_col_chunk_size because the number of cols
+      // will not always be evenly divided by m_col_chunk_size. In most cases, the
+      // extra space will not cause any problems, but it does sometimes.
+      auto p_lay_k           = subview_2dkc(d_pmid, m_nlay);
```

Comment:
This code is worth looking at since it shows how well we were able to leverage the existing fields.